caluma/caluma_form/domain_logic.py

```diff
@@ -98,12 +98,29 @@ def set_file(validated_data):
 class SaveDocumentLogic:
     @staticmethod
     @transaction.atomic
-    def create(
-        # form: models.Form, meta: Optional[dict] = None, user: Optional[BaseUser] = None
-        **kwargs,
-    ):
-        # return BaseLogic.create(models.Document, form=form, meta=meta, user=user)
-        return BaseLogic.create(models.Document, **kwargs)
+    def create(**kwargs):
+        document = BaseLogic.create(models.Document, **kwargs)
+
+        for question in document.form.questions.filter(
+            default_answer__isnull=False
+        ).iterator():
```

Comment:
Hm, What is the `.iterator()` here for?

Reply:
Please correct my if I'm wrong:

We only need to evaluate those queries once. When using `iterator()` in combination with a postgresql DB, we're disabling caching of the result on QuerySet level and use a server-side cursor to fetch the records one-by-one form the DB. From the [docs](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#iterator):

> For a QuerySet which returns a large number of objects that you only need to access once, this can result in better performance and a significant reduction in memory.

Reply:
Awesome! Today I learned!